gradle/generation/local-settings.gradle

```diff
@@ -35,26 +35,48 @@ configure(rootProject) {
         def testsJvms = (int) Math.max(1d, Math.min(cpus * 0.5d, 12))
 
         // Write the defaults for this machine.
-        rootProject.file("gradle.properties").write(
-            [
-                "# These settings have been generated automatically on the first run.",
-                "# See gradlew :helpLocalSettings for more information.",
-                "systemProp.file.encoding=UTF-8",
-                "org.gradle.jvmargs=-Xmx3g", // TODO figure out why "gradlew check" runs out of memory if 2g
-                "org.gradle.parallel=true",
-                "org.gradle.priority=normal",
-                "org.gradle.warning.mode=none", // Silence gradle warnings. We'll deal with them when we upgrade the wrapper.
-                "",
-                "# You may disable the background daemon if it consumes too much memory.",
-                "org.gradle.daemon=true",
-                "org.gradle.daemon.idletimeout=900000", // timeout after 15 mins.
-                "",
-                "# Maximum number of parallel gradle workers.",
-                "org.gradle.workers.max=${maxWorkers}",
-                "",
-                "# Maximum number of test JVMs forked per test task.",
-                "tests.jvms=${testsJvms}"
-            ].join("\n"), "UTF-8")
+        rootProject.file("gradle.properties").write("""
```

Comment:
Do we need some kind of flag to regenerate this file if it exists but users are upgrading from an older gradle/jvm version?

Reply:
I wondered about this too, but in the end opted for simplicity. This initial generated settings file is meant to provide the defaults and pointers about where other things can be tweaked. But it is meant to be tweaked to your (the developer's) needs and I know some folks are doing just that. Manipulating their changes would only create more headaches. You can always wipe this file and let it regenerate if you want it refreshed entirely.

I think these days you could also use init scripts to set up your local Solr builds completely independently from the repository structure [1]. I haven't toyed with this though.

https://docs.gradle.org/current/userguide/init_scripts.html#sec:using_an_init_script